### PR TITLE
MAISTRA-2478 Honor enableIngressClassName in ingress status syncer

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

**Comment by luksa** [comment]
@shamsher31 After this is merged, you can revert the following change in the operator: https://github.com/maistra/istio-operator/commit/c2c78966f39ce507fccc02d04d3f0b318ef082ea#diff-c5f5b082b87676e0375856cd9a705a245141c481b76b35546eda9c0487646ac6R234-R237

File: pilot/pkg/config/kube/ingress/status_test.go [hunk 1]
```diff
@@ -115,7 +115,7 @@ func makeStatusSyncer(t *testing.T) *StatusSyncer {
 
 	client := kubelib.NewFakeClient()
 	setupFake(t, client)
-	sync := NewStatusSyncer(fakeMeshHolder("istio-ingress"), client, false)
+	sync := NewStatusSyncer(fakeMeshHolder("istio-ingress"), client, false, true)
```
Comment (shamsher31) [inline pilot/pkg/config/kube/ingress/status_test.go:118]: Does this means `IngressClassName` will be enabled by default?

Comment (shamsher31) [inline pilot/pkg/config/kube/ingress/status_test.go:118]: `IngressClassName` is disabled by default right?

Comment (luksa) [reply pilot/pkg/config/kube/ingress/status_test.go:118]: No. In Istio itself, it's enabled, but we disable it in Maistra via `enableIngressClassName=false`. Here, it has to be enabled for the unit tests to pass.

**Review by shamsher31 — APPROVED** [review]

**Review by brian-avery — APPROVED** [review]

**Comment by brian-avery** [comment]
@luksa is this ready for an `ok-to-merge`?

**Pull request merged**